modules/copy-post.php

```diff
@@ -0,0 +1,310 @@
+	}
+
+	/**
+	 * Update the target post's Likes and Sharing statuses.
+	 *
+	 * @param WP_Post $source_post Post object to be copied.
+	 * @param int     $target_post_id Target post ID.
+	 * @return array Array with the results of each update action.
+	 */
+	protected function update_likes_sharing( $source_post, $target_post_id ) {
+		$likes          = get_post_meta( $source_post->ID, 'switch_like_status', true );
+		$sharing        = get_post_meta( $source_post->ID, 'sharing_disabled', false );
+		$likes_result   = update_post_meta( $target_post_id, 'switch_like_status', $likes );
+		$sharing_result = update_post_meta( $target_post_id, 'sharing_disabled', $sharing );
+		return array(
+			'likes'   => $likes_result,
+			'sharing' => $sharing_result,
+		);
+	}
+
+	/**
+	 * Update the target post's title.
+	 *
+	 * @param string  $post_title Post title determined by `get_default_post_to_edit()`.
+	 * @param WP_Post $post       Post object of newly-inserted post.
+	 * @return string             Updated post title from source post.
+	 */
+	public function filter_title( $post_title, $post ) {
+		return $post->post_title;
+	}
+
+	/**
+	 * Update the target post's content (`post_content`).
+	 *
+	 * @param string  $post_content Post content determined by `get_default_post_to_edit()`.
+	 * @param WP_Post $post         Post object of newly-inserted post.
+	 * @return string               Updated post content from source post.
+	 */
+	public function filter_content( $post_content, $post ) {
+		return $post->post_content;
+	}
+
+	/**
+	 * Update the target post's excerpt.
+	 *
+	 * @param string  $post_excerpt Post excerpt determined by `get_default_post_to_edit()`.
+	 * @param WP_Post $post         Post object of newly-inserted post.
+	 * @return string               Updated post excerpt from source post.
+	 */
+	public function filter_excerpt( $post_excerpt, $post ) {
+		return $post->post_excerpt;
+	}
+
+	/**
+	 * Validate the post type to be used for the target post.
+	 *
+	 * @param WP_Post $post Post object of current post in listing.
+	 * @return bool True if the post type is in a list of supported psot types; false otherwise.
+	 */
+	protected function validate_post_type( $post ) {
+		$valid_post_types    = array(
+			'post',
+			'page',
+			'jetpack-testimonial',
+			'jetpack-portfolio',
+		);
+		$post_type_supported = in_array( $post->post_type, $valid_post_types, true );
+
+		/**
+		 * Fires when determining if the "Copy" row action should be made available.
+		 * Allows overriding supported post types.
+		 *
+		 * @module copy-post
+		 *
+		 * @since 7.0
+		 *
+		 * @param bool  $post_type_supported If the given post type is a valid supported psot type.
+		 * @param array $valid_post_types Supported post types.
+		 * @param WP_Post $post Post object of current post in listing.
+		 */
+		return apply_filters( 'jetpack_copy_post_post_types', $post_type_supported, $valid_post_types, $post );
```

Comment:
How about moving this filter a bit above, so `$valid_post_types` can be filtered? It seems that it would then be easier to use that way?

Reply:
I liked the idea of having it after, so that you could use it for any other kind of logic too, but you're right, that is a little more confusing and muddying the purpose of the method. So even better, I've filtered on `valid_post_types` as you suggested, and added a new filter on the return of the `add_row_action` method (which gives a point for unique checks or anything else before outputting the Copy action).